### ISSUE TITLE
SSR fails in Gatsby build

### ISSUE THREAD

**Issue by pszafer** [srigar/multiselect-react-dropdown#47]
Hi,

I'm trying to use your component, but it is failing during build in Gatsby.

```
error "window" is not available during server side rendering.
ReferenceError: window is not defined
```
Should I use lib such as `@loadable/component` or is there something to make module working in SSR?

**Comment by srigar** [comment]
@pszafer are u sure is this caused by 'multiselect-react-dropdown', because we are not accessing window object. Pls cross check it.

**Comment by srigar** [comment]
Will look into this.

**Comment by yfchiuaa** [comment]
> Facing the same issue with Next js :
> 
> ```
> Server Error
> ReferenceError: window is not defined
> 
> This error happened while generating the page. Any console logs will be displayed in the terminal window.
> Call Stack
> t
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:1987)
> B
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:2075)
> A.exports
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:5620)
> Object.<anonymous>
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:1052)
> s
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:124)
> Module.<anonymous>
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:14426)
> s
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:124)
> <unknown>
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:923)
> Object.<anonymous>
> file:///Users/ghassen/Documents/hiveway/mondaycar-app/node_modules/multiselect-react-dropdown/build/index.js (1:932)
> Module._compile
> internal/modules/cjs/loader.js (1138:30)
> Object.Module._extensions..js
> internal/modules/cjs/loader.js (1158:10)
> Module.load
> internal/modules/cjs/loader.js (986:32)
> Function.Module._load
> internal/modules/cjs/loader.js (879:14)
> Module.require
> internal/modules/cjs/loader.js (1026:19)
> require
> internal/modules/cjs/helpers.js (72:18)
> eval
> external%20%22multiselect-react-dropdown%22 (1:0)
> ```

For the next.js, I can solve it by using dynamic import like this:
```javascript
import dynamic from 'next/dynamic';
const Multiselect = dynamic(
    () => import('multiselect-react-dropdown').then(module => module.Multiselect),
    {
        ssr: false
    }
)
```

Hope it helps

**Comment by ghassenchaar96** [comment]
Thanks for your response @yfchiuaa .
But still, I'm having an issue with ref. When I do multiselectRef.current.onRemoveSelectedItem() I get TypeError: multiselectRef.current.onRemoveSelectedItem is not a function.
This works perfectly fine on a React app but not in my next app.
console.log(mdEditor.current) shows {retry: ƒ}

console.log(mdEditor.current) shows {retry: ƒ} 🤔

**Comment by Nikhil-devloper** [comment]
How to Write the same code in CRA (Create react app) version?

**Comment by wajeshubham** [comment]
Any updates on this issue? @srigar

**Comment by srigar** [comment]
Its fixed in version >=2.0.0. Feel free to reopen issue if still persist
